src/main/java/org/folio/fqm/service/FqlToSqlConverterService.java

```diff
@@ -68,145 +72,168 @@ public Condition getSqlCondition(String fqlCriteria, EntityType entityType) {
   /**
    * Converts the given FQL condition to the corresponding SQL query
    */
-  public Condition getSqlCondition(FqlCondition<?> fqlCondition, EntityType entityType) {
-    return sqlConverters.getOrDefault(fqlCondition.getClass(), (c, e) -> falseCondition()).apply(fqlCondition, entityType);
+  // The unchecked warning is suppressed because sqlConverters has to have a raw type, forcing its use to have an unchecked call
+  @SuppressWarnings("unchecked")
+  public static Condition getSqlCondition(FqlCondition<?> fqlCondition, EntityType entityType) {
+    final Field<Object> field = fqlCondition instanceof FieldCondition<?> fieldCondition
+      ? field(fieldCondition, entityType)
+      : null;
+    return sqlConverters.getOrDefault(fqlCondition.getClass(), (condition, et, f) -> falseCondition()).handle(fqlCondition, entityType, field);
   }
 
-  private Condition handleEquals(EqualsCondition equalsCondition, EntityType entityType) {
+  private static Condition handleEquals(EqualsCondition equalsCondition, EntityType entityType, Field<Object> field) {
```

Comment:
This is the other tiny change that makes this PR kind of obnoxious: I made almost everything `static`. It's much more efficient and generally makes things a little easier to understand (there's no state outside the method to worry about coming from external dependencies)